### PR TITLE
Address remaining concerns from the 2nd DataFrame API Review 

### PR DESCRIPTION
**Pull request by pgovind**
All of these changes are from the API review. Pretty straightforward

### CODE REVIEW

File: src/Microsoft.Data.Analysis/DataFrame.BinaryOperators.tt [hunk 1]
```diff
@@ -21,18 +21,12 @@ namespace Microsoft.Data.Analysis
...
-        {
-            return df.<#=method.MethodName#>(value);
-        }
-
-        public static DataFrame operator <#=method.Operator#>(bool value, DataFrame df)
-        {
-            return df.Reverse<#=method.MethodName#>(value, inPlace: false);
-        }
-
+<#      // There seem to be very few use cases for an API such as df.And/Or/Xor(true). #>
```
Comment (pgovind) [inline src/Microsoft.Data.Analysis/DataFrame.BinaryOperators.tt]: Maybe replace very few with no use cases

File: src/Microsoft.Data.Analysis/DataFrame.BinaryOperators.cs [hunk 2]
```diff
@@ -32,16 +22,6 @@ public partial class DataFrame
             return df.ReverseAdd(value);
         }
 
-        public static DataFrame operator +(DataFrame df, char value)
```
Comment (eerhardt) [inline src/Microsoft.Data.Analysis/DataFrame.BinaryOperators.cs]: For conversational sake - could this be useful for appending a character to string columns? Or would they just use a full string in that case?

File: src/Microsoft.Data.Analysis/DataFrame.BinaryOperators.cs [hunk 3]
```diff
@@ -662,36 +562,6 @@ public partial class DataFrame
             return df.ReverseModulo(value);
         }
 
-        public static DataFrame operator &(DataFrame df, bool value)
```
Comment (eerhardt) [inline src/Microsoft.Data.Analysis/DataFrame.BinaryOperators.cs]: Why are these being removed?

**Comment by eerhardt** [comment]
>         /// Clips values beyond the specified thresholds on numeric columns

Should the summary be updated to not say "clip" anymore?

[https://docs.microsoft.com/en-us/dotnet/api/system.math.clamp?view=netcore-3.1](https://docs.microsoft.com/en-us/dotnet/api/system.math.clamp?view=netcore-3.1)

Uses the summary:

> Returns value clamped to the inclusive range of min and max.

---
Refers to: src/Microsoft.Data.Analysis/DataFrame.cs:251 in 16dc6ea. [](commit_id = 16dc6ea2108e1d64eed3426a430f973cd08a7a26, deletion_comment = False)

File: src/Microsoft.Data.Analysis/DataFrame.cs [hunk 4]
```diff
@@ -278,15 +253,15 @@ public DataFrame OrderByDescending(string columnName)
         /// <typeparam name="U"></typeparam>
         /// <param name="lower">Minimum value. All values below this threshold will be set to it</param>
         /// <param name="upper">Maximum value. All values above this threshold will be set to it</param>
-        public DataFrame Clip<U>(U lower, U upper, bool inPlace = false)
+        public DataFrame Clamp<U>(U lower, U upper, bool inPlace = false)
```
Comment (eerhardt) [inline src/Microsoft.Data.Analysis/DataFrame.cs]: Maybe `lower` => `min` and `upper` => `max`?

File: tests/Microsoft.Data.Analysis.Tests/DataFrameTests.cs [hunk 5]
```diff
@@ -1654,30 +1606,30 @@ public void TestMerge()
...
+            Assert.Null( merge.Columns["Int_left"][12]);
+            Assert.Null( merge.Columns["Int_left"][5]);
             VerifyMerge(merge, left, right, JoinAlgorithm.FullOuter);
 
             // Inner merge 
             merge = left.Merge<int>(right, "Int", "Int", joinAlgorithm: JoinAlgorithm.Inner);
             Assert.Equal(9, merge.Rows.Count);
             Assert.Equal(merge.Columns.Count, left.Columns.Count + right.Columns.Count);
-            Assert.Equal(merge["Int_right"][2], right["Int"][2]);
+            Assert.Equal( merge.Columns["Int_right"][2], right.Columns["Int"][2]);
```
Comment (eerhardt) [inline tests/Microsoft.Data.Analysis.Tests/DataFrameTests.cs]: Are there unnecessary whitespace changes here?

File: src/Microsoft.Data.Analysis/DataFrame.BinaryOperators.cs (hunk 2 above)
Comment (pgovind) [reply src/Microsoft.Data.Analysis/DataFrame.BinaryOperators.cs]: 2 things:
1. This operator is on the DataFrame itself. So it'd be useful only if we had a DataFrame composed of only char columns. Even then, the current behavior is not great: 'a' + 'a' results in whatever character `194` is. 
2. Assuming you mean the same operator but defined on the `DataFrameColumn` type, then no, it wouldn't be useful. We'd just use the full `StringDataFrameColumn` in that case. We don't have the ability to go from a `PrimitiveDataFrameColumn<char>` -> `StringDataFrameColumn` yet.

File: src/Microsoft.Data.Analysis/DataFrame.BinaryOperators.cs (hunk 3 above)
Comment (pgovind) [reply src/Microsoft.Data.Analysis/DataFrame.BinaryOperators.cs]: Because it'd only be useful when we have a `DataFrame` composed of only Boolean columns(because `&/|/^ ABoolValue` don't make sense on any other column). I think the feeling in the API review was that it didn't make sense to keep this API for a corner case. The operators are still defined on the base `DataFrameColumn` type though, so a statement like `boolColumn & true` is valid.

File: src/Microsoft.Data.Analysis/DataFrame.cs (hunk 4 above)
Comment (pgovind) [reply src/Microsoft.Data.Analysis/DataFrame.cs]: Yup. Changed the parameter names in the `Filter` too

File: tests/Microsoft.Data.Analysis.Tests/DataFrameTests.cs (hunk 5 above)
Comment (pgovind) [reply tests/Microsoft.Data.Analysis.Tests/DataFrameTests.cs]: Yup. Search and replace artifact. Good catch!
